src/cpu/x64/jit_uni_reorder.cpp

```diff
@@ -1281,12 +1618,164 @@ struct jit_uni_reorder_t : public primitive_t {
+                const memory_desc_t *src_md, engine_t *dst_engine,
+                const memory_desc_t *dst_md) {
+            auto prb = tr::prb_t();
+
+            status_t prb_init_status = prb_init(prb, *src_md, *dst_md, attr);
+            if (prb_init_status != status::success) return prb_init_status;
+
+            DEBUG({
+                printf("init : ");
+                prb_dump(prb);
+            });
+            // Sort the prb array in increasing sizes of the output stride
+            prb_normalize(prb);
+            DEBUG({
+                printf("norm : ");
+                prb_dump(prb);
+            });
+            /* Combine the variables, which appear together on both
+             * sides of the reorder */
+            prb_simplify(prb);
+            DEBUG({
+                printf("smpl : ");
+                prb_dump(prb);
+            });
+            prb_tile_normalize(prb);
+            DEBUG({
+                printf("smpl : ");
+                prb_dump(prb);
+            });
+
+            if (!tr::jit_single_blk_kernel::applicable(prb)) {
+                return status::unimplemented;
+            }
+
+            auto _pd = new pd_t(attr, src_engine->kind(), src_md,
+                    dst_engine->kind(), dst_md);
+            if (_pd == nullptr) return status::out_of_memory;
+            if (_pd->init(engine, src_engine, dst_engine) != status::success) {
+                delete _pd;
+                return status::unimplemented;
+            }
+            _pd->prb_ = prb;
+            _pd->init_scratchpad_md();
+            return safe_ptr_assign<reorder_pd_t>(*reorder_pd, _pd);
+        }
+
+        tr::prb_t prb_;
+    };
+
+    jit_blk_reorder_t(const pd_t *apd) : primitive_t(apd) {
+        kernel_ = utils::make_unique<tr::jit_single_blk_kernel>(pd()->prb_);
+    }
+
+    inline int n(int d) const {
+        assert(d < pd()->prb_.ndims);
+        return (int)pd()->prb_.nodes[d].n;
+    }
+    inline int is(int d) const {
+        assert(d < pd()->prb_.ndims);
+        return (int)pd()->prb_.nodes[d].is;
+    }
+    inline int os(int d) const {
+        assert(d < pd()->prb_.ndims);
+        return (int)pd()->prb_.nodes[d].os;
+    }
+    inline int ss(int d) const {
+        assert(d < pd()->prb_.ndims);
+        return (int)pd()->prb_.nodes[d].ss;
+    }
+
+    status_t execute(const exec_ctx_t &ctx) const override {
+        auto in = CTX_IN_MEM(const char *, DNNL_ARG_FROM);
+        auto out = CTX_OUT_MEM(char *, DNNL_ARG_TO);
+        DEFINE_SCALES_BUFFER(scales);
+
+        // kernel handle 2-dimension tiles
+        // Which means there is a tail
+        auto &prb = this->pd()->prb_;
+        ptrdiff_t BH = 1;
+        for (int i = 2; i < prb.ndims; ++i) {
+            BH *= prb.nodes[i].n;
+        }
+
+        auto block_sz = n(0);
+        auto FL = (n(1) + block_sz - 1) / block_sz;
+        auto bh_stride = BH == 1 ? 0 /* don't care, and no is(2) */ : is(2);
```

Comment:
is(2) could be greater than 2^31, so please either extend the `is_applicable()` to avoid such cases, or replace int with dim_t (same as int64_t)